src/services/series.service.ts

```diff
@@ -0,0 +1,23 @@
+import { addQueryParams } from '../utils/formatting';
+import { getDataOrThrow } from '../utils/api';
+import { API_BASE_URL } from '../config';
+
+import type { Series, GetSeriesParams } from '#types/series';
+
+/**
+ * Get series by id
+ * @param {string} id
+ * @param params
+ */
+export const getSeries = async (id?: string, params: GetSeriesParams = {}): Promise<Series | undefined> => {
+  if (!id) {
+    return undefined;
+  }
+
+  const pathname = `/apps/series/${id}`;
+  const url = addQueryParams(`${API_BASE_URL}${pathname}`, params);
+  const response = await fetch(url);
+  const data = await getDataOrThrow(response);
```

Comment:
Instead of using getDataOrThrow and adding all the ApiError and response type, why not just check `response.status` here and return undefined if it equals 404?  This function already potentially returns undefined and it prevents using error throwing and relying on error types and inspecting return codes in hooks to control business logic.

Reply:
Hm, returning undefined for such cases doesn't look to me as a good solution. Having an error type provides additional typings for hooks and can promote further customisations. In case we want enrich error object / add code, we can do it in the constructor. Right now we can't even get the code from api response in the hooks. We can move the business logic to the hook but I am not sure that service in the right place.